eras/conway/impl/src/Cardano/Ledger/Conway/Delegation/Certificates.hs

```diff
@@ -0,0 +1,102 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE LambdaCase #-}
+
+module Cardano.Ledger.Conway.Delegation.Certificates
+  ( ConwayDCert (..),
+    transDCert,
+  )
+where
+
+import Cardano.Ledger.BaseTypes (invalidKey)
+import Cardano.Ledger.Binary
+  ( FromCBOR (..),
+    FromCBORGroup (..),
+    ToCBOR (..),
+    ToCBORGroup (..),
+    decodeRecordSum,
+    encodeListLen,
+    listLenInt,
+  )
+import Cardano.Ledger.Crypto
+import Cardano.Ledger.Shelley.Delegation.Certificates (DCert (..), DelegCert (..), Delegation (..), GenesisDelegCert (..), PoolCert (..))
+import Cardano.Ledger.Slot (EpochNo (..))
+import Control.DeepSeq (NFData)
+import Data.Word (Word8)
+import GHC.Generics (Generic)
+import NoThunks.Class (NoThunks)
+
+data ConwayDCert c
+  = ConwayDCertDeleg !(DelegCert c)
+  | ConwayDCertPool !(PoolCert c)
+  | ConwayDCertGenesis !(GenesisDelegCert c)
```

Comment:
```suggestion
  | ConwayDCertConstitutional !(ConstitutionalDelegCert c)
```

(I'm not expert at naming things, but I think it would be great to drop the "genesis" moniker)